atc/db/build_in_memory_check.go

```diff
@@ -206,6 +206,7 @@ func (b *inMemoryCheckBuildForApi) Events(from uint) (EventSource, error) {
 					err = psql.Select("1").
 						From("resources").
 						Where(sq.Eq{"id": b.resourceId}).
+						Where(sq.Eq{"in_memory_build_status": BuildStatusStarted}).
```

Comment:
Without this line, the build events of  a resource that is with var source error will not be closed from server, so UI will keep showing `loading ...` . The reason is no resource config scope was created due to var error and this query will actually find one record, thus it will return completed as false.
https://github.com/concourse/concourse/blob/4e6ad40006517d4dcf61cc8146125bf82831456f/atc/db/build_in_memory_check.go#L215